Build failure: python313Packages.pymupdf

### Nixpkgs version

- Unstable (25.11)

### Steps to reproduce

Build the package.

### Can Hydra reproduce this build failure?

Hydra is currently rebuilding this package.

### Link to Hydra build job

_No response_

### Relevant log output

```console
Executing pytestCheckPhase
pytest flags: -m pytest --deselect=tests/test_tables.py::test_markdown -k not\ \(test_codespell\)\ and\ not\ \(test_pylint\)\ and\ not\ \(test_flake8\)\ and\ not\ \(test_color_count\)\ and\ not\ \(test_3050\)\ and\ not\ \(test_textbox3\)\ and\ not\ \(test_3493\)\ and\ not\ \(test_4180\)\ and\ not\ \(test_4457\)\ and\ not\ \(test_4445\)\ and\ not\ \(test_4533\)\ and\ not\ \(test_open2\)
[1m============================= test session starts ==============================[0m
platform linux -- Python 3.13.9, pytest-8.4.2, pluggy-1.6.0
rootdir: /build/source
configfile: pytest.ini
[1mcollecting ... [0m[1m
collecting 236 items                                                           [0m[1m
collected 378 items / 13 deselected / 365 selected                             [0m

tests/test_2548.py [32m.[0m[33m                                                     [  0%][0m
tests/test_2634.py [32m.[0m[33m                                                     [  0%][0m
tests/test_2904.py [32m.[0m[33m                                                     [  0%][0m
tests/test_2907.py [32m.[0m[33m                                                     [  1%][0m
tests/test_4141.py [32m.[0m[33m                                                     [  1%][0m
tests/test_4503.py [32m.[0m[33m                                                     [  1%][0m
tests/test_4505.py [32m.[0m[33m                                                     [  1%][0m
tests/test_4520.py [32m.[0m[33m                                                     [  2%][0m
tests/test_4614.py [32m.[0m[33m                                                     [  2%][0m
tests/test_annots.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                 [ 12%][0m
tests/test_badfonts.py [32m.[0m[33m                                                 [ 12%][0m
tests/test_balance_count.py [32m.[0m[33m                                            [ 12%][0m
tests/test_barcode.py [32m.[0m[33m                                                  [ 12%][0m
tests/test_clip_page.py [32m.[0m[33m                                                [ 13%][0m
tests/test_cluster_drawings.py [32m.[0m[32m.[0m[32m.[0m[33m                                       [ 13%][0m
tests/test_crypting.py [32m.[0m[33m                                                 [ 14%][0m
tests/test_docs_samples.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                              [ 18%][0m
tests/test_drawings.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                          [ 20%][0m
tests/test_embeddedfiles.py [32m.[0m[32m.[0m[33m                                           [ 21%][0m
tests/test_extractimage.py [32m.[0m[32m.[0m[32m.[0m[33m                                           [ 22%][0m
tests/test_font.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                           [ 25%][0m
tests/test_general.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m [ 38%]
[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                   [ 49%][0m
tests/test_geometry.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                        [ 52%][0m
tests/test_imagebbox.py [32m.[0m[32m.[0m[33m                                               [ 52%][0m
tests/test_imagemasks.py [32m.[0m[32m.[0m[33m                                              [ 53%][0m
tests/test_import.py [32m.[0m[33m                                                   [ 53%][0m
tests/test_insertimage.py [32m.[0m[32m.[0m[32m.[0m[33m                                            [ 54%][0m
tests/test_insertpdf.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                       [ 56%][0m
tests/test_linebreaks.py [32m.[0m[33m                                               [ 57%][0m
tests/test_linequad.py [32m.[0m[33m                                                 [ 57%][0m
tests/test_memory.py [32m.[0m[32m.[0m[32m.[0m[33m                                                 [ 58%][0m
tests/test_metadata.py [32m.[0m[32m.[0m[32m.[0m[33m                                               [ 59%][0m
tests/test_mupdf_regressions.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                    [ 60%][0m
tests/test_named_links.py [32m.[0m[32m.[0m[32m.[0m[33m                                            [ 61%][0m
tests/test_nonpdf.py [32m.[0m[32m.[0m[32m.[0m[33m                                                 [ 62%][0m
tests/test_object_manipulation.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                   [ 63%][0m
tests/test_objectstreams.py [32m.[0m[32m.[0m[32m.[0m[33m                                          [ 64%][0m
tests/test_optional_content.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                      [ 65%][0m
tests/test_page_links.py [32m.[0m[33m                                               [ 65%][0m
tests/test_pagedelete.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                            [ 66%][0m
tests/test_pagelabels.py [32m.[0m[32m.[0m[33m                                              [ 67%][0m
tests/test_pixmap.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                          [ 74%][0m
tests/test_remove-rotation.py [32m.[0m[33m                                          [ 74%][0m
tests/test_rewrite_images.py [32m.[0m[33m                                           [ 74%][0m
tests/test_rtl.py [32m.[0m[33m                                                      [ 75%][0m
tests/test_showpdfpage.py [32m.[0m[32m.[0m[33m                                             [ 75%][0m
tests/test_spikes.py [32m.[0m[33m                                                   [ 75%][0m
tests/test_story.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[33m                                               [ 77%][0m
tests/test_tables.py [32m.[0m[32m.[0m[32m.[0m[31mF[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                                     [ 81%][0m
tests/test_tesseract.py [32m.[0m[32m.[0m[32m.[0m[31m                                              [ 82%][0m
tests/test_textbox.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                                        [ 85%][0m
tests/test_textextract.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31mE[0m[32m.[0m[31mF[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31mF[0m[31mE[0m[32m.[0m[31m                      [ 91%][0m
tests/test_textsearch.py [32m.[0m[32m.[0m[32m.[0m[31m                                             [ 92%][0m
tests/test_toc.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                                           [ 95%][0m
tests/test_widgets.py [32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[32m.[0m[31m                                     [ 99%][0m
tests/test_word_delimiters.py [32m.[0m[31m                                          [100%][0m

==================================== ERRORS ====================================
[31m[1m________________________ ERROR at teardown of test_3594 ________________________[0m

request = <SubRequest 'wrap' for <Function test_3594>>

    [0m[37m@pytest[39;49;00m.fixture(autouse=[94mTrue[39;49;00m)[90m[39;49;00m
    [94mdef[39;49;00m[90m [39;49;00m[92mwrap[39;49;00m(request):[90m[39;49;00m
    [90m    [39;49;00m[33m'''[39;49;00m
    [33m    Check that tests return with empty MuPDF warnings buffer. For example this[39;49;00m
    [33m    detects failure to call fz_close_output() before fz_drop_output(), which[39;49;00m
    [33m    (as of 2024-4-12) generates a warning from MuPDF.[39;49;00m
    [33m[39;49;00m
    [33m    As of 2024-09-12 we also detect whether tests leave fds open; but for now[39;49;00m
    [33m    do not fail tests, because many tests need fixing.[39;49;00m
    [33m    '''[39;49;00m[90m[39;49;00m
        [94mglobal[39;49;00m PYMUPDF_PYTEST_RESUME[90m[39;49;00m
        [94mif[39;49;00m PYMUPDF_PYTEST_RESUME:[90m[39;49;00m
            [90m# Skip all tests until we reach a matching name.[39;49;00m[90m[39;49;00m
            [94mif[39;49;00m PYMUPDF_PYTEST_RESUME == request.function.[91m__name__[39;49;00m:[90m[39;49;00m
                [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33m### [39;49;00m[33m{[39;49;00mPYMUPDF_PYTEST_RESUME[33m=}[39;49;00m[33m: resuming at [39;49;00m[33m{[39;49;00mrequest.function.[91m__name__[39;49;00m[33m=}[39;49;00m[33m.[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
                PYMUPDF_PYTEST_RESUME = [94mNone[39;49;00m[90m[39;49;00m
            [94melse[39;49;00m:[90m[39;49;00m
                [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33m### [39;49;00m[33m{[39;49;00mPYMUPDF_PYTEST_RESUME[33m=}[39;49;00m[33m: Skipping [39;49;00m[33m{[39;49;00mrequest.function.[91m__name__[39;49;00m[33m=}[39;49;00m[33m.[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
                [94mreturn[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        wt = pymupdf.TOOLS.mupdf_warnings()[90m[39;49;00m
        [94massert[39;49;00m [95mnot[39;49;00m wt, [33mf[39;49;00m[33m'[39;49;00m[33m{[39;49;00mwt[33m=}[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        [94mif[39;49;00m platform.python_implementation() == [33m'[39;49;00m[33mGraalVM[39;49;00m[33m'[39;49;00m:[90m[39;49;00m
            pymupdf.TOOLS.set_small_glyph_heights()[90m[39;49;00m
        [94melse[39;49;00m:[90m[39;49;00m
            [94massert[39;49;00m [95mnot[39;49;00m pymupdf.TOOLS.set_small_glyph_heights()[90m[39;49;00m
        next_fd_before = os.open([91m__file__[39;49;00m, os.O_RDONLY)[90m[39;49;00m
        os.close(next_fd_before)[90m[39;49;00m
    [90m[39;49;00m
        [94mif[39;49;00m platform.system() == [33m'[39;49;00m[33mLinux[39;49;00m[33m'[39;49;00m [95mand[39;49;00m platform.python_implementation() != [33m'[39;49;00m[33mGraalVM[39;49;00m[33m'[39;49;00m:[90m[39;49;00m
            test_fds = [94mTrue[39;49;00m[90m[39;49;00m
        [94melse[39;49;00m:[90m[39;49;00m
            test_fds = [94mFalse[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [94mif[39;49;00m test_fds:[90m[39;49;00m
            [90m# Gather detailed information about leaked fds.[39;49;00m[90m[39;49;00m
            [94mdef[39;49;00m[90m [39;49;00m[92mget_fds[39;49;00m():[90m[39;49;00m
                [94mimport[39;49;00m[90m [39;49;00m[04m[96msubprocess[39;49;00m[90m[39;49;00m
                path = [33m'[39;49;00m[33mPyMuPDF-linx-fds[39;49;00m[33m'[39;49;00m[90m[39;49;00m
                path_l = [33m'[39;49;00m[33mPyMuPDF-linx-fds-l[39;49;00m[33m'[39;49;00m[90m[39;49;00m
                command = [33mf[39;49;00m[33m'[39;49;00m[33mls /proc/[39;49;00m[33m{[39;49;00mos.getpid()[33m}[39;49;00m[33m/fd > [39;49;00m[33m{[39;49;00mpath[33m}[39;49;00m[33m'[39;49;00m[90m[39;49;00m
                command_l = [33mf[39;49;00m[33m'[39;49;00m[33mls -l /proc/[39;49;00m[33m{[39;49;00mos.getpid()[33m}[39;49;00m[33m/fd > [39;49;00m[33m{[39;49;00mpath_l[33m}[39;49;00m[33m'[39;49;00m[90m[39;49;00m
                subprocess.run(command, shell=[94m1[39;49;00m)[90m[39;49;00m
                subprocess.run(command_l, shell=[94m1[39;49;00m)[90m[39;49;00m
                [94mwith[39;49;00m [96mopen[39;49;00m(path) [94mas[39;49;00m f:[90m[39;49;00m
                    ret = f.read()[90m[39;49;00m
                    ret = ret.replace([33m'[39;49;00m[33m\n[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33m [39;49;00m[33m'[39;49;00m)[90m[39;49;00m
                [94mwith[39;49;00m [96mopen[39;49;00m(path_l) [94mas[39;49;00m f:[90m[39;49;00m
                    ret_l = f.read()[90m[39;49;00m
                [94mreturn[39;49;00m ret, ret_l[90m[39;49;00m
            open_fds_before, open_fds_before_l = get_fds()[90m[39;49;00m
    [90m[39;49;00m
        pymupdf._log_items_clear()[90m[39;49;00m
        pymupdf._log_items_active([94mTrue[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        JM_annot_id_stem = pymupdf.JM_annot_id_stem[90m[39;49;00m
    [90m[39;49;00m
        [94mdef[39;49;00m[90m [39;49;00m[92mget_members[39;49;00m(a):[90m[39;49;00m
            ret = [96mdict[39;49;00m()[90m[39;49;00m
            [94mfor[39;49;00m n [95min[39;49;00m [96mdir[39;49;00m(a):[90m[39;49;00m
                [94mif[39;49;00m [95mnot[39;49;00m n.startswith([33m'[39;49;00m[33m_[39;49;00m[33m'[39;49;00m):[90m[39;49;00m
                    v = [96mgetattr[39;49;00m(a, n)[90m[39;49;00m
                    ret[n] = v[90m[39;49;00m
            [94mreturn[39;49;00m ret[90m[39;49;00m
    [90m[39;49;00m
        [90m# Allow post-test checking that pymupdf._globals has not changed.[39;49;00m[90m[39;49;00m
        _globals_pre = get_members(pymupdf._globals)[90m[39;49;00m
    [90m[39;49;00m
        [90m# Run the test.[39;49;00m[90m[39;49;00m
        rep = [94myield[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        sys.stdout.flush()[90m[39;49;00m
    [90m[39;49;00m
        [90m# Test has run; check it did not create any MuPDF warnings etc.[39;49;00m[90m[39;49;00m
        wt = pymupdf.TOOLS.mupdf_warnings()[90m[39;49;00m
        [94mif[39;49;00m [95mnot[39;49;00m [96mhasattr[39;49;00m(pymupdf, [33m'[39;49;00m[33mmupdf[39;49;00m[33m'[39;49;00m):[90m[39;49;00m
            [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33mNot checking mupdf_warnings on classic.[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        [94melse[39;49;00m:[90m[39;49;00m
>           [94massert[39;49;00m [95mnot[39;49;00m wt, [33mf[39;49;00m[33m'[39;49;00m[33mWarnings text not empty: [39;49;00m[33m{[39;49;00mwt[33m=}[39;49;00m[33m'[39;49;00m[90m[39;49;00m
[1m[31mE           AssertionError: Warnings text not empty: wt='Actualtext with no position. Text may be lost or mispositioned.\n... repeated 2 times...'[0m
[1m[31mE           assert not 'Actualtext with no position. Text may be lost or mispositioned.\n... repeated 2 times...'[0m

[1m[31mtests/conftest.py[0m:90: AssertionError
----------------------------- Captured stdout call -----------------------------

Page 0:
Page 1:
Page 2:
Page 3:
Page 4:
Page 5:
Page 6:
Page 7:
Page 8:
Page 9:
Page 10:
Page 11:
Page 12:
Page 13:
Page 14:
Page 15:
Page 16:
Page 17:
Page 18:
Page 19:
Page 20:
Page 21:
Page 22:
Page 23:
Page 24:
Page 25:
Page 26:
Page 27:
Page 28:
Page 29:
Page 30:
Page 31:
Page 32:
Page 33:
Page 34:
Page 35:
Page 36:
Page 37:
Page 38:
Page 39:
Page 40:
Page 41:
Page 42:
Page 43:
Page 44:
Page 45:
Page 46:
Page 47:
Page 48:
Page 49:
Page 50:
Page 51:
Page 52:
Page 53:
Page 54:
Page 55:
Page 56:
Page 57:
Page 58:
Page 59:
Page 60:
Page 61:
Page 62:
Page 63:
Page 64:
Page 65:
Page 66:
Page 67:
Page 68:
Page 69:
Page 70:
Page 71:
Page 72:
Page 73:
Page 74:
Page 75:
Page 76:
Page 77:
Page 78:
Page 79:
Page 80:
Page 81:
[31m[1m________________________ ERROR at teardown of test_4546 ________________________[0m

request = <SubRequest 'wrap' for <Function test_4546>>

    [0m[37m@pytest[39;49;00m.fixture(autouse=[94mTrue[39;49;00m)[90m[39;49;00m
    [94mdef[39;49;00m[90m [39;49;00m[92mwrap[39;49;00m(request):[90m[39;49;00m
    [90m    [39;49;00m[33m'''[39;49;00m
    [33m    Check that tests return with empty MuPDF warnings buffer. For example this[39;49;00m
    [33m    detects failure to call fz_close_output() before fz_drop_output(), which[39;49;00m
    [33m    (as of 2024-4-12) generates a warning from MuPDF.[39;49;00m
    [33m[39;49;00m
    [33m    As of 2024-09-12 we also detect whether tests leave fds open; but for now[39;49;00m
    [33m    do not fail tests, because many tests need fixing.[39;49;00m
    [33m    '''[39;49;00m[90m[39;49;00m
        [94mglobal[39;49;00m PYMUPDF_PYTEST_RESUME[90m[39;49;00m
        [94mif[39;49;00m PYMUPDF_PYTEST_RESUME:[90m[39;49;00m
            [90m# Skip all tests until we reach a matching name.[39;49;00m[90m[39;49;00m
            [94mif[39;49;00m PYMUPDF_PYTEST_RESUME == request.function.[91m__name__[39;49;00m:[90m[39;49;00m
                [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33m### [39;49;00m[33m{[39;49;00mPYMUPDF_PYTEST_RESUME[33m=}[39;49;00m[33m: resuming at [39;49;00m[33m{[39;49;00mrequest.function.[91m__name__[39;49;00m[33m=}[39;49;00m[33m.[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
                PYMUPDF_PYTEST_RESUME = [94mNone[39;49;00m[90m[39;49;00m
            [94melse[39;49;00m:[90m[39;49;00m
                [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33m### [39;49;00m[33m{[39;49;00mPYMUPDF_PYTEST_RESUME[33m=}[39;49;00m[33m: Skipping [39;49;00m[33m{[39;49;00mrequest.function.[91m__name__[39;49;00m[33m=}[39;49;00m[33m.[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
                [94mreturn[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        wt = pymupdf.TOOLS.mupdf_warnings()[90m[39;49;00m
        [94massert[39;49;00m [95mnot[39;49;00m wt, [33mf[39;49;00m[33m'[39;49;00m[33m{[39;49;00mwt[33m=}[39;49;00m[33m'[39;49;00m[90m[39;49;00m
        [94mif[39;49;00m platform.python_implementation() == [33m'[39;49;00m[33mGraalVM[39;49;00m[33m'[39;49;00m:[90m[39;49;00m
            pymupdf.TOOLS.set_small_glyph_heights()[90m[39;49;00m
        [94melse[39;49;00m:[90m[39;49;00m
            [94massert[39;49;00m [95mnot[39;49;00m pymupdf.TOOLS.set_small_glyph_heights()[90m[39;49;00m
        next_fd_before = os.open([91m__file__[39;49;00m, os.O_RDONLY)[90m[39;49;00m
        os.close(next_fd_before)[90m[39;49;00m
    [90m[39;49;00m
        [94mif[39;49;00m platform.system() == [33m'[39;49;00m[33mLinux[39;49;00m[33m'[39;49;00m [95mand[39;49;00m platform.python_implementation() != [33m'[39;49;00m[33mGraalVM[39;49;00m[33m'[39;49;00m:[90m[39;49;00m
            test_fds = [94mTrue[39;49;00m[90m[39;49;00m
        [94melse[39;49;00m:[90m[39;49;00m
            test_fds = [94mFalse[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [94mif[39;49;00m test_fds:[90m[39;49;00m
            [90m# Gather detailed information about leaked fds.[39;49;00m[90m[39;49;00m
            [94mdef[39;49;00m[90m [39;49;00m[92mget_fds[39;49;00m():[90m[39;49;00m
                [94mimport[39;49;00m[90m [39;49;00m[04m[96msubprocess[39;49;00m[90m[39;49;00m
                path = [33m'[39;49;00m[33mPyMuPDF-linx-fds[39;49;00m[33m'[39;49;00m[90m[39;49;00m
                path_l = [33m'[39;49;00m[33mPyMuPDF-linx-fds-l[39;49;00m[33m'[39;49;00m[90m[39;49;00m
                command = [33mf[39;49;00m[33m'[39;49;00m[33mls /proc/[39;49;00m[33m{[39;49;00mos.getpid()[33m}[39;49;00m[33m/fd > [39;49;00m[33m{[39;49;00mpath[33m}[39;49;00m[33m'[39;49;00m[90m[39;49;00m
                command_l = [33mf[39;49;00m[33m'[39;49;00m[33mls -l /proc/[39;49;00m[33m{[39;49;00mos.getpid()[33m}[39;49;00m[33m/fd > [39;49;00m[33m{[39;49;00mpath_l[33m}[39;49;00m[33m'[39;49;00m[90m[39;49;00m
                subprocess.run(command, shell=[94m1[39;49;00m)[90m[39;49;00m
                subprocess.run(command_l, shell=[94m1[39;49;00m)[90m[39;49;00m
                [94mwith[39;49;00m [96mopen[39;49;00m(path) [94mas[39;49;00m f:[90m[39;49;00m
                    ret = f.read()[90m[39;49;00m
                    ret = ret.replace([33m'[39;49;00m[33m\n[39;49;00m[33m'[39;49;00m, [33m'[39;49;00m[33m [39;49;00m[33m'[39;49;00m)[90m[39;49;00m
                [94mwith[39;49;00m [96mopen[39;49;00m(path_l) [94mas[39;49;00m f:[90m[39;49;00m
                    ret_l = f.read()[90m[39;49;00m
                [94mreturn[39;49;00m ret, ret_l[90m[39;49;00m
            open_fds_before, open_fds_before_l = get_fds()[90m[39;49;00m
    [90m[39;49;00m
        pymupdf._log_items_clear()[90m[39;49;00m
        pymupdf._log_items_active([94mTrue[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        JM_annot_id_stem = pymupdf.JM_annot_id_stem[90m[39;49;00m
    [90m[39;49;00m
        [94mdef[39;49;00m[90m [39;49;00m[92mget_members[39;49;00m(a):[90m[39;49;00m
            ret = [96mdict[39;49;00m()[90m[39;49;00m
            [94mfor[39;49;00m n [95min[39;49;00m [96mdir[39;49;00m(a):[90m[39;49;00m
                [94mif[39;49;00m [95mnot[39;49;00m n.startswith([33m'[39;49;00m[33m_[39;49;00m[33m'[39;49;00m):[90m[39;49;00m
                    v = [96mgetattr[39;49;00m(a, n)[90m[39;49;00m
                    ret[n] = v[90m[39;49;00m
            [94mreturn[39;49;00m ret[90m[39;49;00m
    [90m[39;49;00m
        [90m# Allow post-test checking that pymupdf._globals has not changed.[39;49;00m[90m[39;49;00m
        _globals_pre = get_members(pymupdf._globals)[90m[39;49;00m
    [90m[39;49;00m
        [90m# Run the test.[39;49;00m[90m[39;49;00m
        rep = [94myield[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        sys.stdout.flush()[90m[39;49;00m
    [90m[39;49;00m
        [90m# Test has run; check it did not create any MuPDF warnings etc.[39;49;00m[90m[39;49;00m
        wt = pymupdf.TOOLS.mupdf_warnings()[90m[39;49;00m
        [94mif[39;49;00m [95mnot[39;49;00m [96mhasattr[39;49;00m(pymupdf, [33m'[39;49;00m[33mmupdf[39;49;00m[33m'[39;49;00m):[90m[39;49;00m
            [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33mNot checking mupdf_warnings on classic.[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        [94melse[39;49;00m:[90m[39;49;00m
>           [94massert[39;49;00m [95mnot[39;49;00m wt, [33mf[39;49;00m[33m'[39;49;00m[33mWarnings text not empty: [39;49;00m[33m{[39;49;00mwt[33m=}[39;49;00m[33m'[39;49;00m[90m[39;49;00m
[1m[31mE           AssertionError: Warnings text not empty: wt='Actualtext with no position. Text may be lost or mispositioned.\n... repeated 120 times...'[0m
[1m[31mE           assert not 'Actualtext with no position. Text may be lost or mispositioned.\n... repeated 120 times...'[0m

[1m[31mtests/conftest.py[0m:90: AssertionError
----------------------------- Captured stdout call -----------------------------
expected_1_23_5
    JOB No.: 
    Shipper (complete name and address) 发货人(名称及地
    址) 
    SINORICH TRANSPORT LIMITED
    ADD:7C,WEST BLDG.,ZHONGQU
    MANSION,211 ZHONGSHAN
    RD. SHANTOU,515041 CN
    TEL:0754-88570001 FAX:0754-88572709
    S/O No. 
expected_mupdf_1_26_1
    JOB No.: Shipper (complete name and address) 发货人(名称及地址)  Tel:                                  Fax: 
 
    S/O No. 托运单号码     
    SINORICH TRANSPORT LIMITED 
    SHIPPING ORDER 
    托运单 
     市场部: 
    88570009 
    88577019 
    88
pymupdf.version=('1.26.4', '1.26.10', None)
text is:
    JOB No.: 
 
    S/O No. 托运单号码   
    SINORICH TRANSPORT LIMITED 
    SHIPPING ORDER 
    托运单 
     市场部: 
    88570009 
    88577019 
    88572702 
     操作部: 
    88570008 
    88570004 
     文件部: 
    88570003
 
    Notify Party(complete name and address, 
text='JOB No.: \n \nS/O No. 托运单号码   \nSINORICH TRANSPORT LIMITED \nSHIPPING ORDER \n托运单 \n 市场部: \n88570009 \n88577019 \n88572702 \n 操作部: \n88570008 \n88570004 \n 文件部: \n88570003\n \nNotify Party(complete name and address, '
text.encode()=b'JOB No.: \n \nS/O No. \xe6\x89\x98\xe8\xbf\x90\xe5\x8d\x95\xe5\x8f\xb7\xe7\xa0\x81   \nSINORICH TRANSPORT LIMITED \nSHIPPING ORDER \n\xe6\x89\x98\xe8\xbf\x90\xe5\x8d\x95 \n \xe5\xb8\x82\xe5\x9c\xba\xe9\x83\xa8: \n88570009 \n88577019 \n88572702 \n \xe6\x93\x8d\xe4\xbd\x9c\xe9\x83\xa8: \n88570008 \n88570004 \n \xe6\x96\x87\xe4\xbb\xb6\xe9\x83\xa8: \n88570003\n \nNotify Party(complete name and address, '
=================================== FAILURES ===================================
[31m[1m__________________________________ test_2979 ___________________________________[0m

    [0m[94mdef[39;49;00m[90m [39;49;00m[92mtest_2979[39;49;00m():[90m[39;49;00m
    [90m    [39;49;00m[33m"""This tests fix #2979 and #3001.[39;49;00m
    [33m[39;49;00m
    [33m    2979: identical cell count for each row[39;49;00m
    [33m    3001: no change of global glyph heights[39;49;00m
    [33m    """[39;49;00m[90m[39;49;00m
        filename = os.path.join(scriptdir, [33m"[39;49;00m[33mresources[39;49;00m[33m"[39;49;00m, [33m"[39;49;00m[33mtest_2979.pdf[39;49;00m[33m"[39;49;00m)[90m[39;49;00m
        doc = pymupdf.open(filename)[90m[39;49;00m
        page = doc[[94m0[39;49;00m][90m[39;49;00m
        tab = page.find_tables()[[94m0[39;49;00m]  [90m# extract the table[39;49;00m[90m[39;49;00m
        lengths = [96mset[39;49;00m()  [90m# stores all row cell counts[39;49;00m[90m[39;49;00m
        [94mfor[39;49;00m e [95min[39;49;00m tab.extract():[90m[39;49;00m
            lengths.add([96mlen[39;49;00m(e))  [90m# store number of cells for row[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [90m# test 2979[39;49;00m[90m[39;49;00m
        [94massert[39;49;00m [96mlen[39;49;00m(lengths) == [94m1[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        [90m# test 3001[39;49;00m[90m[39;49;00m
        [94massert[39;49;00m ([90m[39;49;00m
            pymupdf.TOOLS.set_small_glyph_heights() [95mis[39;49;00m [94mFalse[39;49;00m[90m[39;49;00m
        ), [33mf[39;49;00m[33m"[39;49;00m[33m{[39;49;00mpymupdf.TOOLS.set_small_glyph_heights()[33m=}[39;49;00m[33m"[39;49;00m[90m[39;49;00m
    [90m[39;49;00m
        wt = pymupdf.TOOLS.mupdf_warnings()[90m[39;49;00m
        [94mif[39;49;00m pymupdf.mupdf_version_tuple >= ([94m1[39;49;00m, [94m26[39;49;00m, [94m0[39;49;00m):[90m[39;49;00m
>           [94massert[39;49;00m ([90m[39;49;00m
                wt[90m[39;49;00m
                == [33m"[39;49;00m[33mbogus font ascent/descent values (3117 / -2463)[39;49;00m[33m\n[39;49;00m[33m... repeated 2 times...[39;49;00m[33m"[39;49;00m[90m[39;49;00m
            )[90m[39;49;00m
[1m[31mE           AssertionError: assert 'bogus font a...d 96 times...' == 'bogus font a...ed 2 times...'[0m
[1m[31mE             [0m
[1m[31mE             Skipping 60 identical leading characters in diff, use -v to show[0m
[1m[31mE             [0m[91m-  2 times...[39;49;00m[90m[39;49;00m[0m
[1m[31mE             [92m+  2 times...[39;49;00m[90m[39;49;00m[0m
[1m[31mE             ?            +[90m[39;49;00m[0m
[1m[31mE             [92m+ Actualtext with no position. Text may be lost or mispositioned.[39;49;00m[90m[39;49;00m[0m
[1m[31mE             [92m+ ... repeated 96 times...[39;49;00m[90m[39;49;00m[0m

[1m[31mtests/test_tables.py[0m:188: AssertionError
--------------------------- Captured stdout teardown ---------------------------
Test has changed process fds:
    open_fds_before='0 1 10 11 12 13 14 2 3 4 5 6 7 8 9 '
     open_fds_after='0 1 10 11 12 13 14 15 2 3 4 5 6 7 8 9 '
open_fds_before_l:
    total 0
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 0 -> /dev/null
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 1 -> /build/#67717915 (deleted)
    l-wx------ 1 nixbld nixbld 64 Nov 14 11:35 10 -> /dev/null
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 11 -> /build/source/tests/resources/image-file1.pdf
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 12 -> /build/source/tests/resources/001003ED.pdf
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 13 -> /build/source/tests/resources/Bezier.epub
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 14 -> /build/source/tests/resources/001003ED.pdf
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 2 -> /build/#67717916 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 3 -> /dev/null
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 4 -> /dev/null
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 5 -> /dev/pts/5
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 6 -> /build/#67717915 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 7 -> /dev/pts/5
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 8 -> /build/#67717916 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 9 -> /dev/pts/5

open_fds_after_l:
    total 0
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 0 -> /dev/null
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 1 -> /build/#67717915 (deleted)
    l-wx------ 1 nixbld nixbld 64 Nov 14 11:35 10 -> /dev/null
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 11 -> /build/source/tests/resources/image-file1.pdf
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 12 -> /build/source/tests/resources/001003ED.pdf
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 13 -> /build/source/tests/resources/Bezier.epub
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 14 -> /build/source/tests/resources/001003ED.pdf
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 15 -> /build/source/tests/resources/test_2979.pdf
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 2 -> /build/#67717916 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 3 -> /dev/null
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 4 -> /dev/null
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 5 -> /dev/pts/5
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 6 -> /build/#67717915 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 7 -> /dev/pts/5
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 8 -> /build/#67717916 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 9 -> /dev/pts/5

Test has leaked fds, next_fd_before=15 next_fd_after=16.
[31m[1m__________________________________ test_3705 ___________________________________[0m

    [0m[94mdef[39;49;00m[90m [39;49;00m[92mtest_3705[39;49;00m():[90m[39;49;00m
        path = os.path.normpath([33mf[39;49;00m[33m'[39;49;00m[33m{[39;49;00m[91m__file__[39;49;00m[33m}[39;49;00m[33m/../../tests/resources/test_3705.pdf[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        [94mdef[39;49;00m[90m [39;49;00m[92mget_all_page_from_pdf[39;49;00m(document, last_page=[94mNone[39;49;00m):[90m[39;49;00m
            [94mif[39;49;00m last_page:[90m[39;49;00m
                document.select([96mlist[39;49;00m([96mrange[39;49;00m([94m0[39;49;00m, last_page)))[90m[39;49;00m
            [94mif[39;49;00m document.page_count > [94m30[39;49;00m:[90m[39;49;00m
                document.select([96mlist[39;49;00m([96mrange[39;49;00m([94m0[39;49;00m, [94m30[39;49;00m)))[90m[39;49;00m
            [94mreturn[39;49;00m [96miter[39;49;00m(page [94mfor[39;49;00m page [95min[39;49;00m document)[90m[39;49;00m
    [90m[39;49;00m
        filename = os.path.basename(path)[90m[39;49;00m
    [90m[39;49;00m
        doc = pymupdf.open(path)[90m[39;49;00m
        texts0 = [96mlist[39;49;00m()[90m[39;49;00m
        [94mfor[39;49;00m i, page [95min[39;49;00m [96menumerate[39;49;00m(get_all_page_from_pdf(doc)):[90m[39;49;00m
            text = page.get_text()[90m[39;49;00m
            [96mprint[39;49;00m(i, text)[90m[39;49;00m
            texts0.append(text)[90m[39;49;00m
    [90m[39;49;00m
        texts1 = [96mlist[39;49;00m()[90m[39;49;00m
        doc = pymupdf.open(path)[90m[39;49;00m
        [94mfor[39;49;00m page [95min[39;49;00m doc:[90m[39;49;00m
            [94mif[39;49;00m page.number >= [94m30[39;49;00m:  [90m# leave the iterator immediately[39;49;00m[90m[39;49;00m
                [94mbreak[39;49;00m[90m[39;49;00m
            text = page.get_text()[90m[39;49;00m
            texts1.append(text)[90m[39;49;00m
    [90m[39;49;00m
        [94massert[39;49;00m texts1 == texts0[90m[39;49;00m
    [90m[39;49;00m
        wt = pymupdf.TOOLS.mupdf_warnings()[90m[39;49;00m
        [94mif[39;49;00m pymupdf.mupdf_version_tuple < ([94m1[39;49;00m, [94m27[39;49;00m):[90m[39;49;00m
>           [94massert[39;49;00m wt == [33m'[39;49;00m[33mActualtext with no position. Text may be lost or mispositioned.[39;49;00m[33m\n[39;49;00m[33m... repeated 434 times...[39;49;00m[33m'[39;49;00m[90m[39;49;00m
[1m[31mE           AssertionError: assert 'Actualtext w...7684 times...' == 'Actualtext w... 434 times...'[0m
[1m[31mE             [0m
[1m[31mE             Skipping 67 identical leading characters in diff, use -v to show[0m
[1m[31mE             [0m[91m-  repeated 434 times...[39;49;00m[90m[39;49;00m[0m
[1m[31mE             ?            --[90m[39;49;00m[0m
[1m[31mE             [92m+  repeated 7684 times...[39;49;00m[90m[39;49;00m[0m
[1m[31mE             ?           +++[90m[39;49;00m[0m

[1m[31mtests/test_textextract.py[0m:379: AssertionError
----------------------------- Captured stdout call -----------------------------
0 
1  
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 
 

2  

3  
• 
• 
• 
• 
• 
• 
• 
• 
 

4  
 
• 
• 
• 

5  
 
 
• 
• 
• 
 
• 
• 
 

6 • 
• 
• 
• 
• 
• 
• 
 
• 
• 
• 
• 
• 
• 
• 
• 
• 
• 

7 • 
 
 
• 
• 
- 
- 
- 
- 
- 
- 
- 
- 

8  
• 
• 
• 
o 
o 
o 
 
 
 

9  
 
• 
• 
• 
• 
• 
• 
• 

10  
 
• 
• 
• 
• 
• 
 

11  
 
 
 
 

12 
13  
 
• 
• 
• 
• 
• 
• 
 

14 • 
• 
• 
• 
• 
• 
• 
• 
• 
• 
 
 
• 
• 

15 • 
• 
• 
• 
• 
• 
• 
• 
• 
 

16 
17  
 
• 
• 
• 
• 
• 

18 ✓
✓
✓
✓
 
 
 

19  
• 
• 
• 
• 
• 
• 
• 
• 
• 
 

20  
 
 

21  
• 
• 
• 
• 
• 
• 

22  

23 • 
• 
 
 

24  

25 
26  

27 
28 
29  

--------------------------- Captured stdout teardown ---------------------------
Test has changed process fds:
    open_fds_before='0 1 10 11 12 13 14 2 3 4 5 6 7 8 9 '
     open_fds_after='0 1 10 11 12 13 14 16 2 3 4 5 6 7 8 9 '
open_fds_before_l:
    total 0
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 0 -> /dev/null
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 1 -> /build/#67717915 (deleted)
    l-wx------ 1 nixbld nixbld 64 Nov 14 11:35 10 -> /dev/null
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 11 -> /build/source/tests/resources/image-file1.pdf
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 12 -> /build/source/tests/resources/001003ED.pdf
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 13 -> /build/source/tests/resources/Bezier.epub
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 14 -> /build/source/tests/resources/001003ED.pdf
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 2 -> /build/#67717916 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 3 -> /dev/null
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 4 -> /dev/null
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 5 -> /dev/pts/5
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 6 -> /build/#67717915 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 7 -> /dev/pts/5
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 8 -> /build/#67717916 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 9 -> /dev/pts/5

open_fds_after_l:
    total 0
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 0 -> /dev/null
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 1 -> /build/#67717915 (deleted)
    l-wx------ 1 nixbld nixbld 64 Nov 14 11:35 10 -> /dev/null
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 11 -> /build/source/tests/resources/image-file1.pdf
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 12 -> /build/source/tests/resources/001003ED.pdf
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 13 -> /build/source/tests/resources/Bezier.epub
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 14 -> /build/source/tests/resources/001003ED.pdf
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 16 -> /build/source/tests/resources/test_3705.pdf
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 2 -> /build/#67717916 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 3 -> /dev/null
    lr-x------ 1 nixbld nixbld 64 Nov 14 11:35 4 -> /dev/null
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 5 -> /dev/pts/5
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 6 -> /build/#67717915 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 7 -> /dev/pts/5
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 8 -> /build/#67717916 (deleted)
    lrwx------ 1 nixbld nixbld 64 Nov 14 11:35 9 -> /dev/pts/5

[31m[1m__________________________________ test_4546 ___________________________________[0m

    [0m[94mdef[39;49;00m[90m [39;49;00m[92mtest_4546[39;49;00m():[90m[39;49;00m
        [90m# This issue will not be fixed (in mupdf) because the test input is faulty.[39;49;00m[90m[39;49;00m
        [90m#[39;49;00m[90m[39;49;00m
        path = os.path.normpath([33mf[39;49;00m[33m'[39;49;00m[33m{[39;49;00m[91m__file__[39;49;00m[33m}[39;49;00m[33m/../../tests/resources/test_4546.pdf[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        [94mwith[39;49;00m pymupdf.open(path) [94mas[39;49;00m document:[90m[39;49;00m
            page = document[[94m0[39;49;00m][90m[39;49;00m
            text = page.get_text()[:[94m200[39;49;00m][90m[39;49;00m
    [90m[39;49;00m
        [90m# We can't actually test with 1.23.5 because it uses `fitz.` not `pymupdf.`.[39;49;00m[90m[39;49;00m
        expected_1_23_5 = [33mb[39;49;00m[33m'[39;49;00m[33mJOB No.: [39;49;00m[33m\n[39;49;00m[33mShipper (complete name and address) [39;49;00m[33m\xe5[39;49;00m[33m\x8f[39;49;00m[33m\x91[39;49;00m[33m\xe8[39;49;00m[33m\xb4[39;49;00m[33m\xa7[39;49;00m[33m\xe4[39;49;00m[33m\xba[39;49;00m[33m\xba[39;49;00m[33m([39;49;00m[33m\xe5[39;49;00m[33m\x90[39;49;00m[33m\x8d[39;49;00m[33m\xe7[39;49;00m[33m\xa7[39;49;00m[33m\xb0[39;49;00m[33m\xe5[39;49;00m[33m\x8f[39;49;00m[33m\x8a[39;49;00m[33m\xe5[39;49;00m[33m\x9c[39;49;00m[33m\xb0[39;49;00m[33m\n[39;49;00m[33m\xe5[39;49;00m[33m\x9d[39;49;00m[33m\x80[39;49;00m[33m) [39;49;00m[33m\n[39;49;00m[33mSINORICH TRANSPORT LIMITED[39;49;00m[33m\n[39;49;00m[33mADD:7C,WEST BLDG.,ZHONGQU[39;49;00m[33m\n[39;49;00m[33mMANSION,211 ZHONGSHAN[39;49;00m[33m\n[39;49;00m[33mRD. SHANTOU,515041 CN[39;49;00m[33m\n[39;49;00m[33mTEL:0754-88570001 FAX:0754-88572709[39;49;00m[33m\n[39;49;00m[33mS/O No. [39;49;00m[33m'[39;49;00m.decode()[90m[39;49;00m
    [90m[39;49;00m
        [90m# This output is different from expected_1_23_5.[39;49;00m[90m[39;49;00m
        expected_mupdf_1_26_1 = [33mb[39;49;00m[33m'[39;49;00m[33mJOB No.: Shipper (complete name and address) [39;49;00m[33m\xe5[39;49;00m[33m\x8f[39;49;00m[33m\x91[39;49;00m[33m\xe8[39;49;00m[33m\xb4[39;49;00m[33m\xa7[39;49;00m[33m\xe4[39;49;00m[33m\xba[39;49;00m[33m\xba[39;49;00m[33m([39;49;00m[33m\xe5[39;49;00m[33m\x90[39;49;00m[33m\x8d[39;49;00m[33m\xe7[39;49;00m[33m\xa7[39;49;00m[33m\xb0[39;49;00m[33m\xe5[39;49;00m[33m\x8f[39;49;00m[33m\x8a[39;49;00m[33m\xe5[39;49;00m[33m\x9c[39;49;00m[33m\xb0[39;49;00m[33m\xe5[39;49;00m[33m\x9d[39;49;00m[33m\x80[39;49;00m[33m)  Tel:                                  Fax: [39;49;00m[33m\n[39;49;00m[33m [39;49;00m[33m\n[39;49;00m[33mS/O No. [39;49;00m[33m\xe6[39;49;00m[33m\x89[39;49;00m[33m\x98[39;49;00m[33m\xe8[39;49;00m[33m\xbf[39;49;00m[33m\x90[39;49;00m[33m\xe5[39;49;00m[33m\x8d[39;49;00m[33m\x95[39;49;00m[33m\xe5[39;49;00m[33m\x8f[39;49;00m[33m\xb7[39;49;00m[33m\xe7[39;49;00m[33m\xa0[39;49;00m[33m\x81[39;49;00m[33m     [39;49;00m[33m\n[39;49;00m[33mSINORICH TRANSPORT LIMITED [39;49;00m[33m\n[39;49;00m[33mSHIPPING ORDER [39;49;00m[33m\n[39;49;00m[33m\xe6[39;49;00m[33m\x89[39;49;00m[33m\x98[39;49;00m[33m\xe8[39;49;00m[33m\xbf[39;49;00m[33m\x90[39;49;00m[33m\xe5[39;49;00m[33m\x8d[39;49;00m[33m\x95[39;49;00m[33m [39;49;00m[33m\n[39;49;00m[33m [39;49;00m[33m\xe5[39;49;00m[33m\xb8[39;49;00m[33m\x82[39;49;00m[33m\xe5[39;49;00m[33m\x9c[39;49;00m[33m\xba[39;49;00m[33m\xe9[39;49;00m[33m\x83[39;49;00m[33m\xa8[39;49;00m[33m: [39;49;00m[33m\n[39;49;00m[33m88570009 [39;49;00m[33m\n[39;49;00m[33m88577019 [39;49;00m[33m\n[39;49;00m[33m88[39;49;00m[33m'[39;49;00m.decode()[90m[39;49;00m
    [90m[39;49;00m
        [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33mexpected_1_23_5[39;49;00m[33m\n[39;49;00m[33m{[39;49;00mtextwrap.indent(expected_1_23_5,[90m [39;49;00m[33m"[39;49;00m[33m    [39;49;00m[33m"[39;49;00m)[33m}[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33mexpected_mupdf_1_26_1[39;49;00m[33m\n[39;49;00m[33m{[39;49;00mtextwrap.indent(expected_mupdf_1_26_1,[90m [39;49;00m[33m"[39;49;00m[33m    [39;49;00m[33m"[39;49;00m)[33m}[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33m{[39;49;00mpymupdf.version[33m=}[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33mtext is:[39;49;00m[33m\n[39;49;00m[33m{[39;49;00mtextwrap.indent(text,[90m [39;49;00m[33m"[39;49;00m[33m    [39;49;00m[33m"[39;49;00m)[33m}[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33m{[39;49;00mtext[33m=}[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
        [96mprint[39;49;00m([33mf[39;49;00m[33m'[39;49;00m[33m{[39;49;00mtext.encode()[33m=}[39;49;00m[33m'[39;49;00m)[90m[39;49;00m
    [90m[39;49;00m
        [94mif[39;49;00m pymupdf.mupdf_version_tuple >= ([94m1[39;49;00m, [94m26[39;49;00m, [94m1[39;49;00m):[90m[39;49;00m
>           [94massert[39;49;00m text == expected_mupdf_1_26_1[90m[39;49;00m
[1m[31mE           AssertionError: assert 'JOB No.: \n ...and address, ' == 'JOB No.: Shi...88577019 \n88'[0m
[1m[31mE             [0m
[1m[31mE             [0m[91m- JOB No.: Shipper (complete name and address) 发货人(名称及地址)  Tel:                                  Fax: [39;49;00m[90m[39;49;00m[0m
[1m[31mE             [92m+ JOB No.: [39;49;00m[90m[39;49;00m[0m
[1m[31mE             [90m [39;49;00m  [90m[39;49;00m[0m
[1m[31mE             [91m- S/O No. 托运单号码     [39;49;00m[90m[39;49;00m[0m
[1m[31mE             ?                 --[90m[39;49;00m[0m
[1m[31mE             [92m+ S/O No. 托运单号码   [39;49;00m[90m[39;49;00m...[0m
[1m[31mE             [0m
[1m[31mE             ...Full output truncated (15 lines hidden), use '-vv' to show[0m

[1m[31mtests/test_textextract.py[0m:891: AssertionError
----------------------------- Captured stdout call -----------------------------
expected_1_23_5
    JOB No.: 
    Shipper (complete name and address) 发货人(名称及地
    址) 
    SINORICH TRANSPORT LIMITED
    ADD:7C,WEST BLDG.,ZHONGQU
    MANSION,211 ZHONGSHAN
    RD. SHANTOU,515041 CN
    TEL:0754-88570001 FAX:0754-88572709
    S/O No. 
expected_mupdf_1_26_1
    JOB No.: Shipper (complete name and address) 发货人(名称及地址)  Tel:                                  Fax: 
 
    S/O No. 托运单号码     
    SINORICH TRANSPORT LIMITED 
    SHIPPING ORDER 
    托运单 
     市场部: 
    88570009 
    88577019 
    88
pymupdf.version=('1.26.4', '1.26.10', None)
text is:
    JOB No.: 
 
    S/O No. 托运单号码   
    SINORICH TRANSPORT LIMITED 
    SHIPPING ORDER 
    托运单 
     市场部: 
    88570009 
    88577019 
    88572702 
     操作部: 
    88570008 
    88570004 
     文件部: 
    88570003
 
    Notify Party(complete name and address, 
text='JOB No.: \n \nS/O No. 托运单号码   \nSINORICH TRANSPORT LIMITED \nSHIPPING ORDER \n托运单 \n 市场部: \n88570009 \n88577019 \n88572702 \n 操作部: \n88570008 \n88570004 \n 文件部: \n88570003\n \nNotify Party(complete name and address, '
text.encode()=b'JOB No.: \n \nS/O No. \xe6\x89\x98\xe8\xbf\x90\xe5\x8d\x95\xe5\x8f\xb7\xe7\xa0\x81   \nSINORICH TRANSPORT LIMITED \nSHIPPING ORDER \n\xe6\x89\x98\xe8\xbf\x90\xe5\x8d\x95 \n \xe5\xb8\x82\xe5\x9c\xba\xe9\x83\xa8: \n88570009 \n88577019 \n88572702 \n \xe6\x93\x8d\xe4\xbd\x9c\xe9\x83\xa8: \n88570008 \n88570004 \n \xe6\x96\x87\xe4\xbb\xb6\xe9\x83\xa8: \n88570003\n \nNotify Party(complete name and address, '
[33m=============================== warnings summary ===============================[0m
<frozen importlib._bootstrap>:488
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyPacked has no __module__ attribute

<frozen importlib._bootstrap>:488
<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type SwigPyObject has no __module__ attribute

<frozen importlib._bootstrap>:488
  <frozen importlib._bootstrap>:488: DeprecationWarning: builtin type swigvarlink has no __module__ attribute

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
[36m[1m=========================== short test summary info ============================[0m
[31mFAILED[0m tests/test_tables.py::[1mtest_2979[0m - AssertionError: assert 'bogus font a...d 96 times...' == 'bogus font a...ed...
[31mFAILED[0m tests/test_textextract.py::[1mtest_3705[0m - AssertionError: assert 'Actualtext w...7684 times...' == 'Actualtext w... 4...
[31mFAILED[0m tests/test_textextract.py::[1mtest_4546[0m - AssertionError: assert 'JOB No.: \n ...and address, ' == 'JOB No.: Shi...88...
[31mERROR[0m tests/test_textextract.py::[1mtest_3594[0m - AssertionError: Warnings text not empty: wt='Actualtext with no position. T...
[31mERROR[0m tests/test_textextract.py::[1mtest_4546[0m - AssertionError: Warnings text not empty: wt='Actualtext with no position. T...
[31m= [31m[1m3 failed[0m, [32m362 passed[0m, [33m13 deselected[0m, [33m5 warnings[0m, [31m[1m2 errors[0m[31m in 109.19s (0:01:49)[0m[31m =[0m
<sys>:0: DeprecationWarning: builtin type swigvarlink has no __module__ attribute
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.57, NixOS, 25.11 (Xantusia), 25.11.20251111.9da7f1c`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/cjjpqrp9xy44j5fjh3mygyxy3i7mslj2-source`

### Notify maintainers



---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/pymupdf/conftest-dont-pip-install.patch
+++ b/pkgs/development/python-modules/pymupdf/conftest-dont-pip-install.patch
@@ -1,0 +1,12 @@
+diff --git a/tests/conftest.py b/tests/conftest.py
+index 8b9bd31b..472f57b8 100644
+--- a/tests/conftest.py
++++ b/tests/conftest.py
+@@ -29,7 +29,6 @@ def install_required_packages():
+     print(f'{__file__}:install_required_packages)(): Running: {command}', flush=1)
+     subprocess.run(command, shell=1, check=1)
+ 
+-install_required_packages()
+ 
+ # Need to import pymupdf only after we've installed pymupdf-fonts above,
+ # because pymupdf imports pymupdf_fonts, and copes with import failure.

--- a/pkgs/development/python-modules/pymupdf/default.nix
+++ b/pkgs/development/python-modules/pymupdf/default.nix
@@ -2,7 +2,6 @@
   lib,
   stdenv,
   buildPythonPackage,
-  pythonOlder,
   fetchFromGitHub,
   python,
   toPythonModule,
@@ -45,22 +44,33 @@ let
 in
 buildPythonPackage rec {
   pname = "pymupdf";
-  version = "1.26.4";
+  version = "1.26.6";
   pyproject = true;
-
-  disabled = pythonOlder "3.9";
 
   src = fetchFromGitHub {
     owner = "pymupdf";
     repo = "PyMuPDF";
     tag = version;
-    hash = "sha256-bzyScV7vznuBQNP8nTjHL2exIs/rVmJBH+soyuAwIGI=";
+    hash = "sha256-CYDgMhsOqqm9AscJxVcjU72P63gpJafj+2cj03RFGaw=";
   };
+
+  patches = [
+    # `conftest.py` tries to run `pip install` to install test dependencies.
+    ./conftest-dont-pip-install.patch
+  ];
 
   # swig is not wrapped as Python package
   postPatch = ''
     substituteInPlace setup.py \
-      --replace-fail "ret.append( 'swig')" "pass" \
+      --replace-fail "ret.append('swig')" "pass" \
+      --replace-fail "ret.append('swig==4.3.1')" "pass"
+  '';
+
+  # `build_extension` passes arguments to `$LD` that are meant for `c++`.
+  # When `LD` is not set, `build_extension` falls back to using `c++` in `PATH`.
+  # See https://github.com/pymupdf/PyMuPDF/blob/1.26.6/pipcl.py#L1998 for details.
+  preConfigure = ''
+    unset LD
   '';
 
   nativeBuildInputs = [
@@ -123,6 +133,7 @@ buildPythonPackage rec {
     "test_4457"
     "test_4445"
     "test_4533"
+    "test_4702"
     # Not a git repository, so git ls-files fails
     "test_open2"
   ];

--- a/pkgs/development/python-modules/pymupdf/default.nix
+++ b/pkgs/development/python-modules/pymupdf/default.nix
@@ -73,11 +73,14 @@ buildPythonPackage rec {
     unset LD
   '';
 
-  nativeBuildInputs = [
+  build-system = [
     libclang
     swig
-    psutil
     setuptools
+  ];
+
+  dependencies = [
+    mupdf-cxx-lib
   ];
 
   buildInputs = [
@@ -88,8 +91,6 @@ buildPythonPackage rec {
     libjpeg_turbo
     gumbo
   ];
-
-  propagatedBuildInputs = [ mupdf-cxx-lib ];
 
   env = {
     # force using system MuPDF (must be defined in environment and empty)
@@ -110,11 +111,9 @@ buildPythonPackage rec {
 
   nativeCheckInputs = [
     pytestCheckHook
-  ];
-
-  checkInputs = [
     fonttools
     pillow
+    psutil
     pymupdf-fonts
   ];
 

